update-result.js

```diff
@@ -0,0 +1,161 @@
+ * and saves the file in place, ready to be reviewed and committed.
+ *
+ * Usage:
+ * updateResult("es5", ["Root test name", "Subtest name"], "graalvm19", "true");
+ *
+ * @param {string} suite Suite name corresponding to results in ./data-{suite}.js
+ * @param {string[]} testPath Array of test names, from top-level to the test to update
+ * @param {string} resultKey Key for the engine/runtime under test. Alphanumeric + underscores
+ * @param {string} result The *raw* string to insert as the value - eg for a fail, use 'false'
+ */
+function updateResult(suite, testPath, resultKey, result) {
+  var suiteFile = path.join(__dirname, "data-" + suite + ".js");
+  var srcString = fs.readFileSync(suiteFile, "utf8");
+  var testNodes = parseSourceForRootTestNodes(srcString);
+  var testNode;
+  var testNameNode;
+  var readableTestPath = [];
+  for (var i = 0; i < testPath.length; i++) {
+    if (!testNodes || testNodes.type !== "ArrayExpression") {
+      throw new Error(
+        "Expected subtests at " +
+          suite +
+          ', test "' +
+          testPath.slice(0, i + 1).join('" / "') +
+          '"'
+      );
+    }
+    testNode = testNodes.elements.find(function (testNode) {
+      return getTestName(testNode) === testPath[i];
+    });
+    if (!testNode) {
+      throw new Error(
+        'No subtest with name "' +
+          testPath[i] +
+          '" in ' +
+          suite +
+          ', test "' +
+          testPath.slice(0, i + 1).join('" / "') +
+          '"'
+      );
+    }
+    testNameNode = maybeGetPropertyValue(testNode, "name");
+    readableTestPath.push(
+      testNameNode ? testNameNode.value : ["MISSING TEST NAME"]
+    );
+    testNodes = maybeGetPropertyValue(testNode, "subtests");
+  }
+  var resNode = maybeGetPropertyValue(testNode, "res");
+  if (!resNode) {
+    throw new Error(
+      'Missing "res" property on "' + readableTestPath.join('" / "') + '"'
+    );
+  }
+
+  // Determine the indentation of the line the object begins (ie, left of "{")
+  var startOfLine = srcString.lastIndexOf("\n", resNode.start) + 1;
+  var indentLength = 0;
+  while (srcString[startOfLine + indentLength] === " ") {
+    indentLength++;
+  }
+
+  var newRawResString = generateNewRawResString(
+    srcString,
+    indentLength,
+    resNode,
+    resultKey,
+    result
+  );
+  fs.writeFileSync(
+    suiteFile,
+    srcString.substring(0, resNode.start) +
+      newRawResString +
+      srcString.substring(resNode.end)
+  );
+}
+
+function generateNewRawResString(
+  srcString,
+  indentLength,
+  resNode,
+  resultKey,
+  result
+) {
+  var newResRaw;
+  var existingValue = maybeGetPropertyValue(resNode, resultKey);
+  if (existingValue) {
+    newResRaw =
+      srcString.substring(resNode.start, existingValue.start) +
+      result +
+      srcString.substring(existingValue.end, resNode.end);
+  } else {
+    var indent = new Array(indentLength).fill(" ").join("");
```

Comment:
`new Array` should never be used; `fill` doesn't untaint a sparse array.
```suggestion
    var indent = Array.from({ length: indentLength }, () => ' ').join('');
```
altho in this case:
```suggestion
    var indent = ' '.repeat(indentLength);
```